docs/developer/inventory.md

```diff
@@ -0,0 +1,164 @@
+| `platform.<platform>` | yes | Define the asset filename for specific platforms. Assets may have different filenames per platform. A fallback `platform.all` key can be used for platform-agnostic assets. | Keys like `platform."windows/amd64"`, `platform."linux/amd64"`, `platform."darwin/arm64"`; **Fallback Key:** `platform.all`. The value is the filename as a string. Templates with `{version}` are allowed. Use `"!tarball"` or `"!zipball"` to get the GitHub-generated source archives. |
+| `executable`          | no  | Indicates whether the downloaded asset should be marked as executable.                                                                                           | `true` or `false` (default).                                                                                                                                                                                                     |
+| `destination`         | yes | The local path where the asset should be saved after download. If the asset is a file, the destination will be its filename. Else, it will be the directory where the contents will be extracted in | Any valid file path string (e.g., `"downloads/asset.exe"`).                                                                                                                                                           |
+| `extract`             | no  | Instructions for file extraction from the downloaded asset.                                                                                                    | `false` (default) for no extraction; a list of glob strings to extract matching files; or a table with keys (see below). |
+| `extract.globs`       | no  | a list of glob strings to match specific files from an archive | Any valid glob such as `*.exe`, `bin/**/asset` ([options](https://docs.python.org/3/library/fnmatch.html)). Will extract all files in the archive if omitted.  |
+| `extract.flatten`     | no  | copy the files to the destination root | `true` or `false` (default) |
+
+## Running the Script
+
+The inventory script supports three commands:
+- `lock`
+- `sync`
+- `list`
+
+The following sections assume you invoke the script with `poetry run`.
+
+#### lock
+
+The `lock` command updates the lock file based on the configuration defined in
+`pyproject.toml` / `inventory.toml`. For each asset, it reads its details,
+queries GitHub to find the appropriate release and asset URL, computes a
+checksum if the asset is available locally, uses caching for fetching and
+hashing, renders filenames that contain the `{version}` template string, and
+supports assets that are platform-agnostic using `platform.all`.
+
+Example:
+
+```
+./dev_scripts/inventory.py lock
+Processing 'asset1'
+Processing 'asset2'
+Lock file 'inventory.lock' updated.
+```
+
+#### sync
```

Comment:
I'm wondering if this `sync` should be actually called `download` instead. I know this is coming from the analogy with `poetry`, but it just seems better to my eyes to have this named differently, as `sync` doesn't really convey it's meaning to non poetry-aware readers. Curious about what you think?

Reply:
I originally called it `install`, would that look better to you? `sync` is probably wrong after all, since we cannot remove assets for other platforms, as is.

Reply:
Yep, install seem a better naming :+1:

Reply:
Fixed in 64afdb1aa116eece6e1e4048bf31467029d1acdc.